build: migrate adev to use ng_project

josephperrott opened the pull request.

josephperrott committed:
build: migrate adev to use ng_project

Move all of adev to use ng_project instead of ng_module

devversion commented on adev/shared-docs/providers/BUILD.bazel:
```diff
@@ -26,10 +25,13 @@ ng_module(
         "//packages/core",
         "//packages/router",
     ],
+    tsconfig = "//adev/shared-docs:tsconfig_build",
```
We probably should also automate this here in `defaults2.bzl`, or follow the pattern in devtools with a dedicated macro file?

devversion commented on adev/shared-docs/services/search.service.ts:
```diff
@@ -10,16 +10,19 @@ import {Injectable, InjectionToken, Provider, inject, resource, signal} from '@a
   SearchResult as AlgoliaSearchResult,
 } from 'algoliasearch/lite';
+// An import from @algolia/client-common is necessary so that the `LiteClient` type is portable.
+import type {} from '@algolia/client-common';
```
Is this necessary? with the explicit type it should be portable at this point

josephperrott replied:
Your right, its correct now, my original change needed these but it can be removed now.

josephperrott committed:
build: move ts_project and ng_project for adev into a macro

Move ts_project and ng_project into a macro for adev/ instead of providing a custom
tsconfig on every target

devversion reviewed (approved):
The stamping change is still confusing to me, but we can follow-up on this as part of the clean-ups.

thePunderWoman commented:
This PR was merged into the repository by commit ef0f1a4ca8d8af11bcedf7874494e4d626e83aa0.

The changes were merged into the following branches: main
